messages/src/lib.rs

```diff
@@ -469,26 +498,25 @@ impl MacAddrs {
     /// that the full range is valid. Specifically, we check that the base MAC
     /// is not a multicast address, that all addresses in the range share the
     /// same OUI, and that count/stride are nonzero.
-    pub const fn new(base_mac: [u8; 6], count: u16, stride: u8) -> Option<Self> {
-        if Self::is_valid(base_mac, count, stride) {
-            Some(Self {
+    pub const fn new(base_mac: [u8; 6], count: u16, stride: u8) -> Result<Self, Error> {
+        match Self::is_valid(base_mac, count, stride) {
```

Comment:
This could be
```rust
Self::is_valid(base_mac, count, stride).map(|_| Self { base_max, count, stride })
```
or even
```rust
Self::is_valid(base_mac, count, stride)?;
Ok(Self { base_max, count, stride }))
```

I don't have strong opinions on which of the three is preferable.

Reply:
(or even folding the validity checks into the constructor, since `is_valid` is not public!)

Reply:
Yeah, I played with that a bunch. Those forms are not const, though, so the constructor can't be. That's not the end of the of the world, of course. I don't really feel strongly either way.

Reply:
Folding the checks in makes this problem go away :) Thanks for the suggestion!